### PR TITLE
WIP: Introduce Jest & PHPUnit tests and run them in CircleCI builds

### PR DESCRIPTION
**Pull request by EvanHerman**
- [x] Lay groundwork for Jest testing
- [x] Run Jest tests in the CircleCI builds and generates a coverage report and uploads it to the build artifacts - [Example](https://383-128991767-gh.circle-artifacts.com/0/jest-coverage/index.html) | [Build](https://circleci.com/gh/godaddy/coblocks/383#artifacts/containers/0)
- [x] Introduce `grunt:eslint` and run it in the `jest` build pipeline - [Example](https://circleci.com/gh/godaddy/coblocks/398)

__________________________

### Notes:
To get some of these eslint tests passing we had to append `UNSAFE_` to a few of the react component lifecycles. These should be migrated to the appropriate react lifecycle methods before React 17, as they will no longer function at that point.
- `componentWillMount` -> `UNSAFE_componentWillMount`
- `componentWillReceiveProps` -> `UNSAFE_componentWillReceiveProps`

**See:** https://github.com/reactjs/react-codemod#rename-unsafe-lifecycles

A few eslint rules were disabled that may negatively effect testing accessibility in future:
- `jsx-a11y/no-noninteractive-tabindex`
- `jsx-a11y/role-has-required-aria-props`
- `jsx-a11y/no-onchange`
- `jsx-a11y/anchor-is-valid`
- `jsx-a11y/label-has-for`
- `jsx-a11y/click-events-have-key-events`
- `jsx-a11y/no-static-element-interactions`

**See:**  https://github.com/godaddy/coblocks/blob/add/jest/.eslintrc.json#L55-L61

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+	],
 	"plugins": [
 		[
 			"@wordpress/babel-plugin-makepot",
 			{
 				"output": "languages/coblocks-js.pot"
 			}
 		]
+	],
+	"test": [
```
Comment (jrtashjian) [inline .babelrc]: I don't think this is supposed to be here. Babel is throwing an error in the build process and indicating `"test"` is not a valid key for `.babelrc`.

**Comment by jrtashjian** [comment]
I'm working through resolving the lint errors to get CI passing.

**Comment by jrtashjian** [comment]
Started with `3007 problems (3007 errors, 0 warnings)` and am down to `22 problems (22 errors, 0 warnings)`.

We should get unit tests added in here before this is merged into master.

**Comment by EvanHerman** [comment]
@jrtashjian Not sure what happened to the Jest tests but its throwing an error now. Seems to be something with the babel version we're using.

**Comment by jrtashjian** [comment]
Because of the huge number of changes in this branch and the need for end-to-end integration testing coming in the near future for blocks, we'll need to test each block manually. Here are the blocks with changes in this PR that we should test thoroughly (all except the Maps block it seems). Please note what exactly is being tested as you go through them so that we have a head start on what integration tests we should write for each block.

- [ ] accordion
- [ ] alert (@paranoia1906)
- [ ] author (@paranoia1906)
- [ ] buttons (@paranoia1906)
- [ ] click-to-tweet
- [ ] dynamic-separator (@paranoia1906)
- [ ] features (@jrtashjian)
- [ ] food-and-drinks (@jrtashjian)
- [ ] form (@EvanHerman)
- [ ] gallery-carousel (@richtabor)
- [ ] gallery-masonry (@richtabor)
- [ ] gallery-stacked (@richtabor)
- [ ] gif
- [ ] gist (@EvanHerman)
- [ ] hero (@richtabor)
- [ ] highlight
- [ ] icon
- [ ] media-card (@jrtashjian)
- [ ] pricing-table (@jrtashjian)
- [ ] row (@EvanHerman)
- [ ] shape-divider
- [ ] share (@EvanHerman)

I assigned 4 to each of us based on what code we've recently touched but please tackle any unassigned afterwards. Just edit this comment and put your name next to the block you're going to test. Create a new issue for each block with your testing with your notes on integration tests we should write.

I think separating these out individually will be simpler than trying to write all tests in the same PR and dealing with merge conflicts.

**Comment by jrtashjian** [comment]
**Edit:** I think what I suggested above is a bit more work than is really necessary.

Let's just go through each block and ensure that things are working properly. We also want to ensure no regressions have been introduced while resolving merge conflicts with the master branch.

**Comment by richtabor** [comment]
@jrtashjian Seems something is up with the Hero block (`CSSGridControl` specifically).

**Comment by richtabor** [comment]
@jrtashjian Shape Divider issue when attempting to resize/save. 
Screen Shot 2019-07-15 at 1 02 50 PM

**Comment by richtabor** [comment]
Gist block hangs up: 
Screen Shot 2019-07-15 at 1 12 26 PM

**Comment by jrtashjian** [comment]
> @jrtashjian Shape Divider issue when attempting to resize/save.
> Screen Shot 2019-07-15 at 1 02 50 PM

Fixed. The `auth_callback` was private rather than public for `register_meta`. Setting to public resolved it.,

**Comment by jrtashjian** [comment]
> * Highlight block breaks when applying a font size via the font size picker in the Settings Sidebar

This is actually a bug on master rather than from the changes here. I've created a new issue for resolving it rather than doing it here.

Issue #667

**Comment by richtabor** [comment]
Good to merge @jrtashjian!

**Pull request merged**